Sources/GRPCHTTP2TransportNIOPosix/HTTP2ServerTransport+Posix.swift

```diff
@@ -21,13 +21,73 @@ import NIOExtras
 import NIOPosix
 
 extension HTTP2ServerTransport {
+  /// A NIOPosix-backed implementation of a server transport.
   @available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
   public struct Posix: ServerTransport {
     private let address: GRPCHTTP2Core.SocketAddress
     private let config: Config
     private let eventLoopGroup: MultiThreadedEventLoopGroup
     private let serverQuiescingHelper: ServerQuiescingHelper
 
+    private enum State {
```

Comment:
No need to undo this but for such a tiny state machine it's totally fine storing the futures/promises inline in the enum cases!

It's also fine to omit the "-State" suffix on the state types because they're already nested within `State` (so are e.g. `State.IdleState`)

Reply:
For a bigger state machine I'd agree, but this has so few states and functions I don't think it's worth it.